### ISSUE TITLE
Support for Tenant level Management REST API in azurerm_api_management

### ISSUE THREAD

**Issue by slynickel** [hashicorp/terraform-provider-azurerm#7435]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Enable the ability to turn on/off the API Management (azurerm_api_management), Management REST API. Visually in the portal it's this setting.

![image](https://user-images.githubusercontent.com/26190757/85349203-7d2dd700-b4c3-11ea-875d-a859511b18a9.png)

In addition to a toggle, retrieving the primary and secondary keys would be required. As far as I can tell both in the portal and SDK documentation the "Identifier" of `integration` cannot be adjusted.

In the Go SDK is [services/apimanagement/mgmt/2019-12-01/apimanagement#TenantAccessClient](
https://godoc.org/github.com/Azure/azure-sdk-for-go/services/apimanagement/mgmt/2019-12-01/apimanagement#TenantAccessClient)

The REST API definitions (I tend to find this a lot easier to read to understand how the Go-SDK actually should be used).
* https://docs.microsoft.com/en-us/rest/api/apimanagement/2019-12-01/tenantaccess/listsecrets
* https://docs.microsoft.com/en-us/rest/api/apimanagement/2019-12-01/tenantaccess/get
* https://docs.microsoft.com/en-us/rest/api/apimanagement/2019-12-01/tenantaccess/update

### New or Affected Resource(s)

* _Existing resource update_ `azurerm_api_management`
   * Adjust to include a setting `management_rest_api_enabled` which would toggle the REST management api.
   * Adjust to include two outputs `management_rest_api_primary_key` and `management_rest_api_secondary_key` that would be marked sensitive.
* _Existing data source update_ `azurerm_api_management`
   * This would be optional, similar adjustments to be made to the data source

Could instead introduce a new resource like `azurerm_api_management_management_api` which is a gross name. This makes less sense to me since the Management REST API applies to the entire APIM resource.

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
resource "azurerm_api_management" "example" {
  name                = "example-apim"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  publisher_name      = "My Company"
  publisher_email     = "company@terraform.io"

  sku_name = "Developer_1"

  management_rest_api_enabled = true # NEW option, should default to false (disabled)

  policy {
    xml_content = <<XML
    <policies>
      <inbound />
      <backend />
      <outbound />
      <on-error />
    </policies>
XML
  }
}

# Need the ability to get both the primary and secondary keys out
output "example1" {
  value = azurerm_api_management.example.management_rest_api_primary_key
}

output "example2" {
  value = azurerm_api_management.example.management_rest_api_secondary_key
}
```

### References

None.

**Comment by meilz381** [comment]
@favoretti I think what is requested here is already implemented. The block is called `tenant_access` and exports the primary key and secondary key.

**Comment by slynickel** [comment]
You are correct. Fixed in #10475
